app/components/download_button_component.rb

```diff
@@ -0,0 +1,29 @@
+class DownloadButtonComponent < ButtonComponent
+  attr_reader :file_data, :file_name, :tag_options
+
+  def initialize(file_data:, file_name:, **tag_options)
+    super(
+      icon: :file_download,
+      action: ->(**tag_options, &block) do
+        link_to(
+          "data:text/plain;charset=utf-8,#{CGI.escape(file_data)}",
```

Comment:
should we make the MIME type configurable? Or in the spirit of YAGNI, add that only if we need something that's not `.txt`?

Reply:
I'd probably call YAGNI on this, in part because I think an implementation could depend on the use-case (for example, automatically detecting MIME type using [magic headers](https://en.wikipedia.org/wiki/List_of_file_signatures)).